src/bin/miri.rs

```diff
@@ -40,9 +36,12 @@ impl<'a> CompilerCalls<'a> for MiriCompilerCalls {
 
             let tcx = state.tcx.unwrap();
             let mir_map = state.mir_map.unwrap();
-            let (span, mir, def_id) = get_main(tcx, mir_map);
+
+            let (node_id, span) = state.session.entry_fn.borrow().expect("no main or start function found");
```

Comment:
Huh, so this returns *either* the `#[main]` or `#[start]` function? That's a bit weird - we need to call them differently. Currently we're calling a `#[start]` without its two arguments, which I believe means they are left as undefined bytes.